### PR TITLE
Stop trying to run services for link_package step

### LINKED ISSUE #14431: SCM/CI link_package + services
The backend changed how it decides if service should run for a package. Services in modes that would not run server side are filtered out a while ago.

We used a project wide service in `localonly` to "trick" the backend into running services for linked packages in the `link_package` step.

https://github.com/openSUSE/open-build-service/blob/672cbc7e8b3608fded117281aa160a7f5c481823/src/api/app/models/workflow/step/link_package_step.rb#L75-L81

This stopped working a while ago, hence services are not running anymore for packages created with the `link_package` step. Which makes them kind of useless for PR events where it's supposed to run the services to pick up changes from the PR...

### PR DESCRIPTION
**Pull request by hennevogel**
Make it possible to combine `branch_package` with `configure_repositories` instead.

Fixes #14431

TODO:
- [x] Write specs
- [x] Write user documentation in the [book](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.obs_workflows.steps.branch_a_package)

### CODE REVIEW

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 1]
```diff
@@ -16,6 +16,16 @@ def target_project_base_name
     step_instructions[:target_project]
   end
 
+  def target_project
+    Project.find_by(name: target_project_name)
+  end
+
+  def add_repositories?
+    return true if step_instructions[:add_repositories].blank?
```
Comment (krauselukas) [inline src/api/app/models/workflow/step/branch_package_step.rb:24]:
```suggestion
    return false if step_instructions[:add_repositories].blank?
```

Comment (krauselukas) [reply src/api/app/models/workflow/step/branch_package_step.rb:24]: @hennevogel I would find it more intuitive to return false if there is nothing to add

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 2]
```diff
@@ -51,6 +61,17 @@ def check_source_access
   def create_branched_package
     check_source_access
 
+    # If we create target_project, BranchPackage.branch below will not create repositories
+    if !add_repositories? && target_project.nil?
```
Comment (krauselukas) [inline src/api/app/models/workflow/step/branch_package_step.rb:65]:
```suggestion
    if add_repositories? && target_project.nil?
```

File: src/api/app/models/workflow/step/branch_package_step.rb (hunk 1 above)
Comment (krauselukas) [reply src/api/app/models/workflow/step/branch_package_step.rb:24]: ah I see, sorry now I get it

File: src/api/app/models/workflow/step/branch_package_step.rb (hunk 2 above)
Comment (krauselukas) [reply src/api/app/models/workflow/step/branch_package_step.rb:65]: nevermind

File: src/api/app/models/workflow/step/branch_package_step.rb (hunk 1 above)
Comment (eduardoj) [reply src/api/app/models/workflow/step/branch_package_step.rb:24]:
```suggestion
    step_instructions[:add_repositories].blank? || step_instructions[:add_repositories] == 'enable'
```
I find this option more intuitive.

**Comment by hennevogel** [comment]
User documentation PR is here now: https://github.com/openSUSE/obs-docu/pull/287

**Review by krauselukas — APPROVED** [review]
Did a quick test, LGTM

**Pull request merged**